Restructure obsv overview (#13119)
kyma-project/kyma: docs/05-technical-reference/00-architecture/obsv-01-architecture-monitoring.md

    >**NOTE:** By default, no notification channels are configured. You need to [set them up](../../03-tutorials/00-observability/obsv-04-send-notifications.md).

6. [**Grafana**](https://grafana.com/docs/guides/getting_started/) provides a dashboard and a graph editor to visualize metrics collected from the Prometheus API. Grafana uses the query language [PromQL](https://prometheus.io/docs/prometheus/latest/querying/basics/) to select and aggregate metrics data from the Prometheus database. Learn how to [access the Grafana UI](../../04-operation-guides/security/sec-06-access-expose-kiali-grafana.md).

## Istio monitoring architecture

The [monitoring chart](https://github.com/kyma-project/kyma/blob/main/resources/monitoring/values.yaml) is preconfigured to collect all metrics relevant for observing the in-cluster [Istio](https://istio.io/latest/docs/concepts/observability/) Service Mesh, including the proxy-level, service-level, and control-plane metrics. 

The whole implementation of our monitoring solution is built around [Istio's observability best practices](https://istio.io/latest/docs/ops/best-practices/observability/).

## Default setup

> ![Prometheus Setup](./assets/obsv-prometheus-setup.svg)

Reviewer: I really like the diagrams you created! They are well aligned with our guidelines 🙂 

One remark: you may want to correct `Istio service mesh` to `Istio Service Mesh`.

Author: Thank you! I will

> 1. The concept of collecting the [service-level](https://istio.io/latest/docs/concepts/observability/#service-level-metrics) metrics is based on the Istio Proxy implemented by Envoy. Istio Proxy collects all communication details inside the service mesh in a decentralized way. After scraping these high-cardinality metrics from the envoys, the metrics must be aggregated on a service level to get the final service-related details.

Reviewer: `service mesh` or `Service Mesh`? 🤔

Reviewer: `After scraping these high-cardinality metrics from the envoys...` - as discussed, I was pretty confused when I saw this word used in this context 😄 Consider explaining what `scraping metrics` means. Or maybe it is sth we expect the user to know?

Author: I'd like to mention the word because it seems common in Prometheus docs, but I agree we should explain it the first time around.

```suggestion
1. The concept of collecting the [service-level](https://istio.io/latest/docs/concepts/observability/#service-level-metrics) metrics is based on the Istio Proxy implemented by Envoy. Istio Proxy collects all communication details inside the service mesh in a decentralized way. After getting (scraping) these high-cardinality metrics from the envoys, the metrics must be aggregated on a service level to get the final service-related details.
```

2. A dedicated Prometheus instance (Prometheus-Istio) scrapes and aggregates the service-level metrics. That instance is configured with the smallest possible data retention time because the raw metrics scraped from the Istio Proxies have high-cardinality and don't need to be kept further. 

> The Istio-Prometheus instance is a Deployment named `monitoring-prometheus-istio-server`, with a hardcoded configuration that must not be changed. It also has no PersistentVolume attached. This instance never discovers additional metric endpoints from such resources as ServiceMonitors.

Reviewer: ```suggestion
The Istio-Prometheus instance is a Deployment named `monitoring-prometheus-istio-server`, with a hardcoded configuration that must not be changed. It also has no Persistent Volume attached. This instance never discovers additional metric endpoints from such resources as Service Monitors.
```

> 3. The main Prometheus instance scrapes these aggregated Istio metrics through the `/federate` endpoint of the Prometheus-Istio instance and any other metric endpoints from such resources as ServiceMonitors.

Reviewer: ```suggestion
3. The main Prometheus instance scrapes these aggregated Istio metrics through the `/federate` endpoint of the Prometheus-Istio instance and any other metric endpoints from such resources as Service Monitors.
```

The main Prometheus instance supports scraping metrics using [`Strict mTLS`](https://istio.io/latest/docs/tasks/security/authentication/authn-policy/#globally-enabling-istio-mutual-tls-in-strict-mode). For this to work, Prometheus is configured to scrape metrics using Istio certificates.  

> 4. Prometheus is deployed with a sidecar proxy which rotates SDS certificates and outputs them to a volume mounted to the corresponding Prometheus container. It is configured to not intercept or redirect any traffic.

Reviewer: ```suggestion
4. Prometheus is deployed with a sidecar proxy which rotates SDS certificates and outputs them to a volume mounted to the corresponding Prometheus container. It is configured to not stop or redirect any traffic.
```

Reviewer: ... or any other simpler word 🤔

Author: I'm not sure which meaning of intercept we're using here :D 
[Merriam-Webster: intercept](https://www.merriam-webster.com/dictionary/intercept) offers several, and not all of them imply _stopping_ the initial movement. If we're in the 1b area, `stop` might not be correct.

> 1a: to stop, seize, or [interrupt](https://www.merriam-webster.com/dictionary/interrupt) in progress or course or before arrival
>  b: to receive (a communication or signal directed elsewhere) usually secretly
> 2a: to gain possession of (an opponent's pass)
>  b: to intercept a pass thrown by (an opponent)
> 3: to include (part of a curve, surface, or solid) between two points, curves, or surfaces
> the part of a circumference intercepted between two radii
> 4 (obsolete) : [PREVENT](https://www.merriam-webster.com/dictionary/prevent), [HINDER](https://www.merriam-webster.com/dictionary/hinder)
> 5 (obsolete): to interrupt communication or connection with

Reviewer 2: "stop" is wrong here. Usually a sidecar is a man in the middle by taking the request, manipulating it and then forwarding it to the application. Here, we want to express that this sidecar is configured to not touch the traffic at all. I think "intercept" is well known and good in this context

> 5. By default, metrics from Kyma components are scraped using mTLS. As an exception, components deployed without sidecar proxy (for example, controllers like Prometheus operator) and Istio system components (for example, the Istio sidecars proxies themselves) are scraped using http; see also [Istio's setup recommendation](https://istio.io/latest/docs/ops/integrations/prometheus/#tls-settings).

Reviewer: ```suggestion
5. By default, metrics from Kyma components are scraped using mTLS. As an exception, components deployed without sidecar proxy (for example, controllers like Prometheus operator) and Istio system components (for example, the Istio sidecars proxies themselves) are scraped using HTTP. For more information, see [Istio's setup recommendation](https://istio.io/latest/docs/ops/integrations/prometheus/#tls-settings).
```

> >**NOTE:** Learn how to [deploy](../../03-tutorials/00-observability/obsv-01-observe-application-metrics.md#deploy-the-example-configuration) a sample `Go` service exposing metrics, which are scraped by Prometheus using mTLS.

Reviewer: ```suggestion
>**NOTE:** Learn how to [deploy a sample `Go` service exposing metrics](../../03-tutorials/00-observability/obsv-01-observe-application-metrics.md#deploy-the-example-configuration) which are scraped by Prometheus using mTLS.
```

### Istio monitoring flow

> The following diagram shows how the Istio-related instance fits into the monitoring setup in Kyma:

Reviewer: ```suggestion
The following diagram shows how the Istio-related instance fits into the Monitoring setup in Kyma:
```

Reviewer: `monitoring` or `Monitoring` ? 🤔

Author: in this case, I think it's a normal noun.

> ![Istio Monitoring](./assets/obsv-monitoring-istio.svg)

Reviewer: Not convinced about putting the resource type into << >> . We'd usually just use `Prometheus Stateful Set`.

Author: Done
